src/XMakeBuildEngine/BackEnd/Node/OutOfProcNode.cs

```diff
@@ -234,15 +234,28 @@ internal static bool IsOutOfProcNode
 
         /// <summary>
         /// Starts up the node and processes messages until the node is requested to shut down.
+        /// Assumes no node reuse.
         /// </summary>
         /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
         /// <returns>The reason for shutting down.</returns>
         public NodeEngineShutdownReason Run(out Exception shutdownException)
         {
+            return Run(false, out shutdownException);
+        }
+
+
+        /// <summary>
+        /// Starts up the node and processes messages until the node is requested to shut down.
+        /// </summary>
+        /// <param name="enableReuse">Whether this node is eligible for reuse later.</param>
+        /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
+        /// <returns>The reason for shutting down.</returns>
+        public NodeEngineShutdownReason Run(bool enableReuse, out Exception shutdownException)
```

Comment:
This gets called at https://github.com/rainersigwald/msbuild/blob/3c015edee424f4eb3535f2235af6ec15f0b57e7f/src/XMakeCommandLine/XMake.cs#L1956.  That'll get set to true when the command line parameter (of the child) is set to true.